### PR TITLE
preventing stack overflow when dropping List.

### PR DESCRIPTION
**Pull request by lulitao1997**
implement our custom Drop trait for List to non-recursively drop the List. #46

### CODE REVIEW

File: src/list/mod.rs [hunk 1]
```diff
@@ -398,6 +398,22 @@ where
     }
 }
 
+impl<T, P> Drop for List<T, P>
```
Comment (orium) [inline src/list/mod.rs]: Add a small comment saying we implement this to avoid a stack overflow when dropping lists.

File: src/list/test.rs [hunk 2]
```diff
@@ -22,6 +22,71 @@ mod iter {
     use super::*;
     use pretty_assertions::assert_eq;
 
+    #[test]
```
Comment (orium) [inline src/list/test.rs]: These tests should probably be before `fn test_serde()`.  Note that this is the `iter` module.

File: src/list/test.rs [hunk 3]
```diff
@@ -22,6 +22,71 @@ mod iter {
...
+    fn test_drop_list() {
+        // When it is dropped, it will set the variable it owned to false.
+        struct DropStruct<'a>(&'a mut bool);
```
Comment (orium) [inline src/list/test.rs]: Can you use a `&'a Cell<bool>` to avoid a `unsafe` code?

**Comment by lulitao1997** [comment]
updated.

**Comment by orium** [comment]
Thanks for contributing!

**Pull request merged**